src/placeos-frontend-loader/loader.cr

```diff
@@ -222,6 +210,17 @@ module PlaceOS::FrontendLoader
         } }
       end
 
+      if repository.deployed_commit_hash != new_commit_hash
+        Log.info { {
+          message:           "updating commit on Repository document",
+          current_commit:    new_commit_hash,
+          repository_commit: repository.commit_hash,
+          folder_name:       repository.folder_name,
+        } }
+        repository.deployed_commit_hash = new_commit_hash
+        repository.update
+      end
```

Comment:
Might be worth returning at the top of this method if the only thing changed on the model is the `deployed_commit_hash`

Reply:
I think it's worth running through everything as we don't know why the user is clicking pull.
maybe an error has occurred and the repo files are missing
also if the commit is set to HEAD then we'll be checking for a new commit.

Reply:
yes, you are right